lib/saml_idp/xml_security.rb

```diff
@@ -53,16 +53,13 @@ def validate(idp_cert_fingerprint, soft = true, options = {})
         log 'Validate the fingerprint'
         base64_cert = find_base64_cert(options)
         cert_text = Base64.decode64(base64_cert)
-        cert =
-          begin
-            OpenSSL::X509::Certificate.new(cert_text)
-          rescue OpenSSL::X509::CertificateError
-            return false if soft
-            raise ValidationError.new(
-              'Invalid certificate',
-              :invalid_certificate_in_request
-            )
-          end
+      begin
+        cert = valid_cert(cert_text)
+      rescue OpenSSL::X509::CertificateError
+        return false if soft
+
+        raise ValidationError.new('Invalid certificate', :invalid_certificate_in_request)
+      end
```

Comment:
WDYT of moving this rescue inside `valid_cert`? could have it return like a `[cert, error]` tuple to check for "soft" validation?


```suggestion
      cert, error = valid_cert(cert_text)
      if error
        if soft
          return false
        else
          raise ValidationError.new('Invalid certificate', :invalid_certificate_in_request)
        end
      end
```

Reply:
ah yeah i didn't like having to repeat it, i'll update

Reply:
updated in https://github.com/18F/saml_idp/pull/126/commits/1eb5cca1e7498f700556ecce5c72989fad66a218